Sources/web3swift/Web3/Web3+GasOracle.swift

```diff
@@ -0,0 +1,214 @@
+                        guard case let .transaction(transaction) = t else { return nil }
+                        return transaction
+                    }
+                }
+                .map { $0.gasPrice }
+
+            return try calculateStatistic(lastNthBlockGasPrice, statistic)
+        }
+    }
+}
+
+public extension Web3.Oracle {
+    // MARK: - Base Fee
+    /// Base fee amount based on last Nth blocks
+    ///
+    /// Normalized means that most high and most low value were droped from calculation.
+    ///
+    /// Nth block may include empty ones.
+    ///
+    /// - Parameter statistic: Statistic to apply for base fee calculation
+    /// - Returns: Suggested base fee amount according to statistic, nil if failed to perdict
+    func predictBaseFee(_ statistic: Statistic) -> BigUInt? {
+        guard let value = try? suggestBaseFee(statistic) else { return nil }
+        return value
+    }
+
+    // MARK: - Tip
+    /// Maximum tip amount based on last block tips
+    ///
+    /// Normalized means that most high and most low value were droped from calculation.
+    ///
+    /// Account first of the latest block that have more than `transactionsNumber` value.
+    ///
+    /// - Parameter statistic: Statistic to apply for tip calculation
+    /// - Returns: Suggested tip amount according to statistic, nil if failed to perdict
+    func predictTip(_ statistic: Statistic) -> BigUInt? {
+        guard let value = try? suggestTipValue(statistic) else { return nil }
+        return value
+    }
+
+    // MARK: - Summary fees
+    /// Method to get summary fees
+    /// - Parameters:
+    ///   - baseFee: Statistic to apply for baseFee
+    ///   - tip: Statistic to apply for tip
+    /// - Returns: Touple where [0] — base fee, [1] — tip, nil if failed to predict
+    func predictBothFees(baseFee: Statistic, tip: Statistic) -> (BigUInt, BigUInt)? {
+        guard let baseFee = try? suggestBaseFee(baseFee) else { return nil }
+        guard let tip = try? suggestTipValue(tip) else { return nil }
+
+        return (baseFee, tip)
+    }
+
+    // MARK: - Legacy GasPrice
+    /// Method to get legacy gas price
+    /// - Parameter statistic: Statistic to apply for gas price
+    /// - Returns: Suggested gas price amount according to statistic, nil if failed to predict
+    func predictGasPriceLegacy(_ statistic: Statistic) -> BigUInt? {
+        guard let value = try? suggestGasFeeLegacy(statistic) else { return nil}
+        return value
+    }
+}
+
+public extension Web3.Oracle {
+    // TODO: Make me struct and incapsulate math within to make me extendable
+    enum Statistic {
+        /// Mininum statistic
+        case minimum
+        /// Mean statistic
+        case mean
+        /// Median statistic
+        case median
+        /// Maximum statistic
+        case maximum
+    }
+}
+
+extension Array {
+    func cropAnomalyValues() -> Self {
+        var tmpArr = self.dropFirst()
```

Comment:
what makes the first and last values anomalous?

Reply:
in statistics the extreme outliers in a set are often looked at to be deviations from the norm, or anomalies. So by clipping them off, you can often get a more meaningful result. In this case the list is sorted,so first and last entries would contain the extreme outliers.